apps/testingaccessibility/src/pages/api/sanity/webhooks/add-all-records-to-algolia.ts

```diff
@@ -18,30 +16,27 @@ export const addAllRecordsToAlgolia = async (
     req,
     res,
   })
-  // Basic security to prevent others from hitting this API
-  const s = get(req.query, 's')
-  if (s !== secret) {
-    res.status(403).json({success: false, message: 'Invalid signature'})
-    return
-  }
 
-  // Fetch the _id of all the documents we want to index
-  const types = ['section', 'lesson']
-  const query = `* [_type in $types][]._id`
+  try {
+    if (!isValidRequest(req, process.env.SANITY_WEBHOOK_SECRET)) {
+      res.status(403).json({success: false, message: 'Invalid signature'})
+    } else {
+      // Fetch the _id of all the documents we want to index
+      const types = ['section', 'lesson']
+      const query = `* [_type in $types][]._id`
 
-  await sanityClient
-    .fetch(query, {types})
-    .then((ids) => {
-      sanityAlgolia.webhookSync(sanityClient, {
-        ids: {created: ids, updated: [], deleted: []},
+      await sanityClient.fetch(query, {types}).then((ids) => {
+        sanityAlgolia.webhookSync(sanityClient as any, {
+          ids: {created: ids, updated: [], deleted: []},
+        })
       })
-    })
-    .then(() => {
+
       res.status(200).send('Success!')
-    })
-    .catch((error) => {
-      res.status(500).json({success: false, message: error})
-    })
+    }
+  } catch (error) {
+    console.error(error)
+    res.status(500).json({success: false, message: error.message})
+  }
```

Comment:
instead of catching in the promise we use the try/catch block around the async/await and capture the error while also writing it to the console on our side.

noticed that in the sanity logs the `message` property was a blank object so here we are responding with the `error.message` so that the Sanity logs will have more info and also writing the full error to our console (it was being swallowed previously)